godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -433,5 +428,24 @@ namespace godel_surface_detection
 
       return name;
     }
+
+    void SurfaceDetection::filterFullCloud()
+    {
+      pcl::PointCloud<pcl::PointXYZRGB>::Ptr intermediate_cloud_ptr(new pcl::PointCloud<pcl::PointXYZRGB>);
+
+      // Create Processing Cloud
+      pcl::VoxelGrid<pcl::PointXYZRGB> vox;
+      vox.setInputCloud (full_cloud_ptr_);
+      vox.setLeafSize (INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE);
+      vox.filter(*intermediate_cloud_ptr);
+
+      pcl::PassThrough<pcl::PointXYZRGB> pass;
```

Comment:
I'd switch around the voxelization and passthrough steps so that we first do pass-through. I think it will produce equivalent data with less overall memory usage.